DataStructures/BigInteger/BigInt.cpp

```diff
@@ -1,14 +1,261 @@
+	aa_test1 *= aa_int_test;
+	std::cout << aa_test1 << '\n';
+	*/
+
+	
+	// COMPLETE
+	// arithmetic-assignment -= test (BigInt and long long params)
+	BigInt aa_test1 = 2;
+	BigInt aa_test2 = 5;
+	int aa_int_test = 10;
+	std::cout << aa_test1 << " set equal to " << aa_test1 << " - " << aa_test2 << " is: ";
+	aa_test1 -= aa_test2;
+	std::cout << aa_test1 << '\n';
+	std::cout << aa_test1 << " set equal to " << aa_test1 << " - " << aa_int_test << " is: ";
+	aa_test1 -= aa_int_test;
+	std::cout << aa_test1 << '\n';
+	
+
+	/*
+	// COMPLETE
+	// relational operators
+	BigInt compare1 = -2;
+	BigInt compare2 = -2;
+	bool result = compare1 == compare2;
+	std::cout << "Is " << compare1 << " equal to " << compare2 << "?: " << result << '\n';
+	compare1 = 1000;
+	int test_int = 100;
+	result = compare1 == test_int;
+	std::cout << "Is " << compare1 << " equal to " << test_int << "?: " << result << '\n';
+	compare1 = -100;
+	compare2 = -100;
+	result = compare1 > compare2;
+	std::cout << "Is " << compare1 << " greater than " << compare2 << "?: " << result << '\n';
+	compare1 = -100001;
+	test_int = -100000;
+	result = compare1 > test_int;
+	std::cout << "Is " << compare1 << " greater than " << test_int << "?: " << result << '\n';
+	compare1 = 100;
+	compare2 = 100;
+	result = compare1 >= compare2;
+	std::cout << "Is " << compare1 << " greater than or equal to " << compare2 << "?: " << result << '\n';
+	compare1 = -100001;
+	test_int = -100001;
+	result = compare1 >= test_int;
+	std::cout << "Is " << compare1 << " greater than or equal to " << test_int << "?: " << result << '\n';
+	compare1 = 1;
+	compare2 = -2;
+	result = compare1 < compare2;
+	std::cout << "Is " << compare1 << " less than " << compare2 << "?: " << result << '\n';
+	compare1 = -100001;
+	test_int = -100000;
+	result = compare1 < test_int;
+	std::cout << "Is " << compare1 << " less than " << test_int << "?: " << result << '\n';
+	compare1 = -2;
+	compare2 = -2;
+	result = compare1 <= compare2;
+	std::cout << "Is " << compare1 << " less than or equal to " << compare2 << "?: " << result << '\n';
+	compare1 = -100001;
+	test_int = -100001;
+	result = compare1 <= test_int;
+	std::cout << "Is " << compare1 << " less than or equal to " << test_int << "?: " << result << '\n';
+	compare1 = -2;
+	compare2 = 2;
+	result = compare1 != compare2;
+	std::cout << "Is " << compare1 << " NOT equal to " << compare2 << "?: " << result << '\n';
+	compare1 = -100001;
+	test_int = 100001;
+	result = compare1 != test_int;
+	std::cout << "Is " << compare1 << " NOT equal to " << test_int << "?: " << result << '\n';
+	*/
+
+	return 0;
+}
+
+
 /*
     Utility functions
     -----------------
 */
 
 bool is_valid_number(const std::string& num) {
     for (char digit : num)
-        if (digit < '0' or digit > '9')
-            return false;
+        if (digit < '0' || digit > '9')
+			if (digit != '+' && digit != '-')
+				return false;
```

Comment:
`is_valid_number()` checks whether the magnitude of the number (which is a string) is a valid number, i.e. it should only have characters from `0` to `9`. So remove the checks for `+` and `-` because it makes the function work incorrectly.